Sources/InfomaniakCore/FileManagement/AppGroupPathProvider.swift

```diff
@@ -0,0 +1,129 @@
+
+    /// The root directory to store the App's Realm files, exists in FS
+    var realmRootURL: URL { get }
+
+    /// The import directory, exists in FS
+    var importDirectoryURL: URL { get }
+
+    /// The cache directory within the app group, exists in FS
+    var cacheDirectoryURL: URL { get }
+
+    /// The temporary directory within the app group, exists in FS
+    var tmpDirectoryURL: URL { get }
+
+    /// Open In Place directory if available
+    var openInPlaceDirectoryURL: URL? { get }
+}
+
+public final class AppGroupPathProvider: AppGroupPathProvidable {
+    private let fileManager = FileManager.default
+
+    private let realmRootPath: String
+
+    // MARK: public var
+
+    public let groupDirectoryURL: URL
+
+    public lazy var realmRootURL: URL = {
+        let drivesURL = groupDirectoryURL.appendingPathComponent(self.realmRootPath, isDirectory: true)
+        try? fileManager.createDirectory(
+            atPath: drivesURL.path,
+            withIntermediateDirectories: true,
+            attributes: nil
+        )
+        return drivesURL
+    }()
+
+    public lazy var importDirectoryURL: URL = {
+        let importURL = groupDirectoryURL.appendingPathComponent("import", isDirectory: true)
+        try? fileManager.createDirectory(
+            atPath: importURL.path,
+            withIntermediateDirectories: true,
+            attributes: nil
+        )
+        return importURL
+    }()
+
+    public lazy var cacheDirectoryURL: URL = {
+        let cacheURL = groupDirectoryURL.appendingPathComponent("Library/Caches", isDirectory: true)
+        try? fileManager.createDirectory(
+            atPath: cacheURL.path,
+            withIntermediateDirectories: true,
+            attributes: nil
+        )
+        return cacheURL
+    }()
+
+    public lazy var tmpDirectoryURL: URL = {
+        let tmpURL = groupDirectoryURL.appendingPathComponent("tmp", isDirectory: true)
+        try? fileManager.createDirectory(
+            atPath: tmpURL.path,
+            withIntermediateDirectories: true,
+            attributes: nil
+        )
+        return tmpURL
+    }()
+
+    public lazy var openInPlaceDirectoryURL: URL? = {
+        let openInPlaceURL = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first?
+            .appendingPathComponent(".shared", isDirectory: true)
+        return openInPlaceURL
+    }()
+
+    // MARK: init
+
+    public init?(realmRootPath: String, appGroupIdentifier: String) {
+        guard let groupDirectoryURL = fileManager.containerURL(forSecurityApplicationGroupIdentifier: appGroupIdentifier)
+        else {
+            return nil
+        }
+
+        do {
+            try fileManager.setAttributes(
+                [FileAttributeKey.protectionKey: FileProtectionType.completeUntilFirstUserAuthentication],
+                ofItemAtPath: groupDirectoryURL.path
+            )
+        } catch {
+            os_log("failed to protect mandatory path")
```

Comment:
If it's already in the dependencies prefer using CocoaLumberjack

Reply:
you are corect it is